kong/api/endpoints.lua

```diff
@@ -241,7 +241,17 @@ local function query_entity(context, self, db, schema, method)
       end
     end
 
-    if key.id and not utils.is_valid_uuid(key.id) then
+    if key.id then
+      if utils.is_valid_uuid(key.id) then
+        local result = is_update and
+                          table.pack(dao[method or context](dao, key, args, opts)) or
+                          table.pack(dao[method or context](dao, key, opts))
+
+        if result[1] or #result > 1 and result[3].code ~= Errors.codes.NOT_FOUND then
+          return table.unpack(result)
+        end
```

Comment:
Perhaps just:
```
local entity, _, err_t = is_update and
                          table.pack(dao[method or context](dao, key, args, opts)) or
                          table.pack(dao[method or context](dao, key, opts))
```

So no ugly table.packing and unpacking.

Reply:
I fixed it the way you suggested but we needed another variable `offset` since it's used in pagination endpoints. I checked the rest of `query_entity` usages and they don't use any additional values. So we should be good with 4: `res, err, err_t, offset`